### ISSUE TITLE
Wine quality prediction using linear regression 

### ISSUE THREAD

**Issue by Saidd16** [Saidd16/Machine-learning-model-to-predict-the-wine-quality-#1]
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error

# Load your dataset into a Pandas DataFrame
# Replace 'your_dataset.csv' with the actual file path to your dataset
data = pd.read_csv('your_dataset.csv')

# Assume 'quality' is the target variable, and other columns are features
X = data.drop('quality', axis=1)
y = data['quality']

# Split the data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Create and train the linear regression model
model = LinearRegression()
model.fit(X_train, y_train)

# Predict on the test set
y_pred = model.predict(X_test)

# Evaluate the model
mse = mean_squared_error(y_test, y_pred)
print('Mean Squared Error:', mse)

# Visualize predictions
plt.scatter(y_test, y_pred)
plt.xlabel('Actual')
plt.ylabel('Predicted')
plt.title('Actual vs Predicted Wine Quality')
plt.show()
